### ISSUE TITLE
Force dissconnect from server due to "Fatally Missing Registry Errors"((Resolved))

### ISSUE THREAD

**Issue by Drakoflame** [VoidSorc/Total-Tinkers#8]
I spent the past two hours stressing out and trying to figure out the issue for why, after adding a few mods, and updating the client modpack, people were unable to join the server, because they were "Fatally Missing Registry Errors", turning into 
![image](https://user-images.githubusercontent.com/8823913/70394680-28ee5a00-19ac-11ea-8ea2-defa16536c86.png)
... _that_... I want to apologize in advanced if I'm coming off aggressively, as I said, I spent two hours trying to find the solution to this issue, and had to resort to removing the new mods, and one-by-one re-adding them and waiting for the issue, only to find a mod that has absolutely 0 in common with wheat, let alone _crafting_, is the one causing people to be unable to log into the server...

[latest.log](https://github.com/UvMidnight/Total-Tinkers/files/3937047/latest.log)

I've attached the latest log from me attempting to join the server. Both client and server are 'default' configs (as I'd thought that maybe it was me enabling the experimental weapon, and making the greatsword core and full guard 'craftable' in the stencil table. I hope the log is enough to help. I will be MORE than glad to help you figure out this issue as soon as humanly possible, I was the one to requested this mod, of all 30 people who've been on the server, we had a whopping 16 **FOR** it, so not having it is a bit of a hit to me because of how I tried selling the mod, and the amount of people who wanted it in return.

**Comment by Drakoflame** [comment]
Any Ideas? I really would like to add this to our server... we have a few people who would like to use something OTHER than a cleaver for a strong weapon

**Comment by Drakoflame** [comment]
I'm bumping this again because I really would like help, even if it isn't this mod that is causing the issue, if we can figure out what's happening so we can use the weapons added by this mod, that would be awesome!

**Comment by Drakoflame** [comment]
Alright it seems like I'm not going to get help for this mod, which really sucks because it looked really fun, and it worked just fine in single player, I greatly enjoyed the weapons, especially the Greatsword, but for now we'll just pass on this mod and maybe revisit it in the future.

I hope that no-one else has this problem, because it was never solved.

**Comment by Drakoflame** [comment]
Alright, I'll admit it, I was a bit of an asshole, and I'm sorry that it happened... and I did the most stupidest thing in the entire world, and didn't even bother to think about it because I was to busy trying to figure out why it was crashing and why... A month later, looking back because we're getting ready to do a major update for our pack and I'm looking at mods that need/can be updated, and what do I notice? Tinkers Construct is now updated to version 2.13.0.171.... we've been using 2.13.0.157 the entire time... and I didn't notice that the version for this mod was updated the same day that Tinkers updated to that version.

I'll bet my money that the issue we were having wasn't some weird issue involving wheat not being able to be crafted from wheat, but an issue that this mod was looking for specific NBT tags, that the server (which had already been running for some time prior to the addition of this mod), did NOT have, while single player _DID_ have it, because it was a fresh install and this mod was able to tell it things upon creation.

Again, I sincerely apologize for my rude behavior and the inexcusable way I acted, I won't give excuses for why I was acting that way, though there may have been IRL issues clouding my thoughts, it's my own damn fault for not checking the fking mod version dependency... The only way I can see a resolve for this issue is to either state in the change logs that it's no longer compatible with older versions/will potentially cause issues on servers.

**Comment by UvMidnight** [comment]
@Drakoflame I am incredibly sorry for not looking at this earlier. I was really not motivated to work on mods during then, and missed the email from github about this issue. Looking at the log, the scimitar was likely the cause of the problem - the log reports that the registry entry for it was missing on the client. 

Either something went wrong with the config and registration chain in the mod's code, or that there was a config mismatch with the server and client. If you have any way of ruling out that there was no config mismatch, it would be great if you could get back to me, it would help out debugging a lot.

**Comment by Drakoflame** [comment]
Yea, I later figured out the issue totally by accident and works now. It turned out that the issue was your mod was the latest version, and we weren't using the latest version of Tinkers. In the latest version they changed how material registers work, and your mod was assuming that they would be read out like that. It worked fine on first load for whatever reason, but when trying to re-connect the server was expecting to tell the client how the materials should look for your weapons, and your weapons were expecting something else, which caused the crash.

I won't pretend to know how it works exactly, but from my experiences with poking at configs and making mods work that weren't intended to work together (I've read a lot of crash logs in my days > 3>;;), I realized that was the issue after we were doing a test pack for a new version with an updated tinkers, and it worked just fine. But for whatever reason it didn't seem to cause the issue in single player XD. 

In short, I don't really think there's anything to fix here, it was simply a case of not updating a mod, which caused issues with addon mods.
